frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -3782,9 +3782,17 @@ const IR::Node* TypeInference::postorder(IR::SwitchStatement* stat) {
         if (isCompileTimeConstant(stat->expression))
             warn(ErrorType::WARN_MISMATCH, "%1%: constant expression in switch", stat->expression);
 
+        std::map<big_int, const IR::Node*> caselabels;
```

Comment:
The type may not be integer: https://p4.org/p4-spec/docs/P4-16-v-1.2.3.html#sec-switch-statement-with-integer-or-enumerated-type-expression

Reply:
I see. I will check and update. Thanks for pointing out!

Reply:
First, I would move this out of typeChecking, it needs to be done only once. Especially since this is a quadratic algorithm and the number of labels can be very large. 
The simplest way would probably be to use the constantFolding pass to generate equality expressions label1 == label2 and see whether any of them evaluates to true.

Reply:
Thanks for the pointers, Mihai. I will explore the constantFolding pass and other possible approaches for detecting this.